src/main/java/org/openrewrite/kotlin/internal/KotlinParserVisitor.java

```diff
@@ -1809,56 +1809,145 @@ public J visitProperty(FirProperty property, ExecutionContext ctx) {
                 }
 
                 int saveCursor = cursor;
-                whitespace();
+                Space s0 = whitespace();
+
+                PsiElement currentPsiElement = getCurrentPsiNode();
+                KtModifierList modifierList = null;
+                List<J.Annotation> annos = new ArrayList<>();
+
+                if (currentPsiElement instanceof KtAnnotationEntry) {
+                    modifierList = PsiTreeUtil.getParentOfType(currentPsiElement, KtModifierList.class);
+                }
+
+                if (modifierList != null) {
+                    List<FirAnnotation> firAnnotations = new ArrayList<>();
+                    firAnnotations.addAll(property.getAnnotations());
+                    firAnnotations.addAll(property.getGetter().getAnnotations());
+                    firAnnotations.addAll(property.getSetter().getAnnotations());
+                    mapModifierList(modifierList, firAnnotations, annos);
+                    if (!annos.isEmpty()) {
+                        J.Annotation a0 = annos.get(0).withPrefix(s0);
```

Comment:
Is there a reason to save s0 and a0 instead of resetting the cursor to savedCursor after the current PsiElement is found?